src/blight/actions/bitcode_extract.py

```diff
@@ -0,0 +1,51 @@
+"""
+The `BitcodeExtract` action.
+"""
+
+import logging
+import os
+import subprocess
+from typing import List
+
+from blight.action import CompilerAction
+from blight.enums import Lang
+from blight.tool import CompilerTool
+
+logger = logging.getLogger(__name__)
+
+
+class BitcodeExtract(CompilerAction):
+    """
+    Compile and extract bitcode. The output bitcode file will be located in the same direcory
+    as the output executable.
+    """
+
+    def before_run(self, tool: CompilerTool) -> None:  # type: ignore
+        store = self._config.get("store")
+        if store is not None:
+            if tool.lang in [Lang.C, Lang.Cxx]:
+                for inpt in tool.inputs:
+                    args: List[str]
+                    if tool.outputs:
+                        args = [
+                            "-c",
+                            "-emit-llvm",
+                            "-o",
+                            store + "/" + inpt.split("/")[-1].split(".")[0] + ".bc",
```

Comment:
You can use `Path` from the standard `pathlib` to make this a little more pleasant:

```python
from pathlib import Path

# ...

store / Path(input).with_suffix(".bc").name
```

More generally though, I _think_ we might want to mangle the names here like we do for `FindOutputs`. Imagine a build like this:

```
src/
    foo.c
    bar.c
    junk/
        stuff.c
        foo.c
```

In this case, there are two separate translation units named `foo.c`, so our current `BitcodeExtract` action would clobber one of them. A reasonable solution might be to take the hash of the input's contents and use that for the bitcode filename; I _think_ that's what GLLVM does in "store" mode.